stdlib/public/core/StringLegacy.swift

```diff
@@ -95,8 +95,33 @@ extension String {
   }
 }
 #else
-// FIXME: Implement hasPrefix and hasSuffix without objc
-// rdar://problem/18878343
+extension String {
+  /// Returns `true` iff `self` begins with `prefix`.
+  public func hasPrefix(prefix: String) -> Bool {
+    if prefix.isEmpty { return false }
```

Comment:
Theoretically *str represent str or <SOME STRING>str that it why I think this should return true. 
Example
--
Given a string s and a literal string "Hello", if I want to check *with hasPrefix*, that adding "Hello" with doesn't changes anything and remains "Hello". 
So, let concStr = "\(s)Hello"; [...] 
conStr.hasPrefix("") shouldn't returns true ?